cli/formatters/environment.go

```diff
@@ -56,11 +59,19 @@ func (f EnvironmentsFormatter) ToStruct(file *file.File) (interface{}, error) {
 	var environmentResource openapi.EnvironmentResource
 	nullableEnvironment := openapi.NewNullableEnvironmentResource(&environmentResource)
 
+	if strings.HasSuffix(file.Path(), ".yaml") || strings.HasSuffix(file.Path(), ".yml") {
```

Comment:
Do we need this validation? Maybe we could try to parse the file, assuming that file definitions *must* be yaml, and throw an error if it's not parseable. Does that make sense?

Reply:
is it guaranteed that the file will always be yaml?

Reply:
A solution (I don't like it much) is to try to parse it as YAML, and then as JSON as a fallback:

```go
func (f EnvironmentsFormatter) ToStruct(file *file.File) (interface{}, error) {
	var environmentResource openapi.EnvironmentResource
	nullableEnvironment := openapi.NewNullableEnvironmentResource(&environmentResource)

	if err := yaml.Unmarshal([]byte(file.Contents()), &environmentResource); err == nil {
		return environmentResource, nil
	}

	err := nullableEnvironment.UnmarshalJSON([]byte(file.Contents()))
	if err != nil {
		return nil, err
	}
	return environmentResource, nil
}
```

Reply:
the original implementation fallbacks to json, and that is also not guaranteed. I think in the end its a product decision: do we want to support files in formats other than YAML?

Reply:
We could assume that it will be YAML and return an error message in case of a parsing error, to avoid having too much parsing logic or guessing on the CLI. Does it make sense?

Reply:
I'm considering the file ContentType. It's hardcoded as YAML. But I still left the json fallback there just in case we change that in the future